File: code/stable/glassbr/src/csharp/InputConstraints.cs
/** \file InputConstraints.cs
    \author Nikitha Krithnan and W. Spencer Smith
    \brief Provides the function for checking the physical constraints and software constraints on the input
*/
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

public class InputConstraints {
    
    /** \brief Verifies that input values satisfy the physical constraints and software constraints
        \param inParams structure holding the input values
    */
    public static void input_constraints(InputParameters inParams) {
        StreamWriter outfile;
        outfile = new StreamWriter("log.txt", true);
        outfile.WriteLine("function input_constraints called with inputs: {");
        outfile.Write("  inParams = ");
        outfile.WriteLine("Instance of InputParameters object");
        outfile.WriteLine("  }");
        outfile.Close();
        
        if (!(0.1 <= inParams.a && inParams.a <= 5.0)) {
            Console.Write("a has value ");
            Console.Write(inParams.a);
            Console.Write(" but expected to be ");
            Console.Write("between ");
            Console.Write(0.1);
            Console.Write(" (d_min)");
            Console.Write(" and ");
            Console.Write(5.0);
            Console.Write(" (d_max)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(0.1 <= inParams.b && inParams.b <= 5.0)) {
            Console.Write("b has value ");
            Console.Write(inParams.b);
            Console.Write(" but expected to be ");
            Console.Write("between ");
            Console.Write(0.1);
            Console.Write(" (d_min)");
            Console.Write(" and ");
            Console.Write(5.0);
            Console.Write(" (d_max)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(4.5 <= inParams.w && inParams.w <= 910.0)) {
            Console.Write("w has value ");
            Console.Write(inParams.w);
            Console.Write(" but expected to be ");
            Console.Write("between ");
            Console.Write(4.5);
            Console.Write(" (w_min)");
            Console.Write(" and ");
            Console.Write(910.0);
            Console.Write(" (w_max)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(6.0 <= inParams.SD && inParams.SD <= 130.0)) {
            Console.Write("SD has value ");
            Console.Write(inParams.SD);
            Console.Write(" but expected to be ");
            Console.Write("between ");
            Console.Write(6.0);
            Console.Write(" (SD_min)");
            Console.Write(" and ");
            Console.Write(130.0);
            Console.Write(" (SD_max)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(inParams.AR <= 5.0)) {
            Console.Write("AR has value ");
            Console.Write(inParams.AR);
            Console.Write(" but expected to be ");
            Console.Write("below ");
            Console.Write(5.0);
            Console.Write(" (AR_max)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        
        if (!(inParams.a > 0)) {
            Console.Write("a has value ");
            Console.Write(inParams.a);
            Console.Write(" but expected to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(inParams.a >= inParams.b)) {
            Console.Write("a has value ");
            Console.Write(inParams.a);
            Console.Write(" but expected to be ");
            Console.Write("above ");
            Console.Write(inParams.b);
            Console.Write(" (b)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(0 < inParams.b && inParams.b <= inParams.a)) {
            Console.Write("b has value ");
            Console.Write(inParams.b);
            Console.Write(" but expected to be ");
            Console.Write("between ");
            Console.Write(0);
            Console.Write(" and ");
            Console.Write(inParams.a);
            Console.Write(" (a)");
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(inParams.w > 0)) {
            Console.Write("w has value ");
            Console.Write(inParams.w);
            Console.Write(" but expected to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(0 < inParams.P_btol && inParams.P_btol < 1)) {
            Console.Write("P_btol has value ");
            Console.Write(inParams.P_btol);
            Console.Write(" but expected to be ");
            Console.Write("between ");
            Console.Write(0);
            Console.Write(" and ");
            Console.Write(1);
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(inParams.TNT > 0)) {
            Console.Write("TNT has value ");
            Console.Write(inParams.TNT);
            Console.Write(" but expected to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(inParams.SD > 0)) {
            Console.Write("SD has value ");
            Console.Write(inParams.SD);
            Console.Write(" but expected to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
        if (!(inParams.AR >= 1)) {
            Console.Write("AR has value ");
            Console.Write(inParams.AR);
            Console.Write(" but expected to be ");
            Console.Write("above ");
            Console.Write(1);
            Console.WriteLine(".");
            throw new Exception("InputError");
        }
    }
}

File: code/stable/glassbr/src/python/InputConstraints.py
## \file InputConstraints.py
# \author Nikitha Krithnan and W. Spencer Smith
# \brief Provides the function for checking the physical constraints and software constraints on the input
from __future__ import print_function
import sys
import math

import InputParameters

## \brief Verifies that input values satisfy the physical constraints and software constraints
# \param inParams structure holding the input values
def input_constraints(inParams):
    outfile = open("log.txt", "a")
    print("function input_constraints called with inputs: {", file=outfile)
    print("  inParams = ", end='', file=outfile)
    print("Instance of InputParameters object", file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    if (not(0.1 <= inParams.a and inParams.a <= 5.0)) :
        print("a has value ", end='')
        print(inParams.a, end='')
        print(" but expected to be ", end='')
        print("between ", end='')
        print(0.1, end='')
        print(" (d_min)", end='')
        print(" and ", end='')
        print(5.0, end='')
        print(" (d_max)", end='')
        print(".")
        raise Exception("InputError")
    if (not(0.1 <= inParams.b and inParams.b <= 5.0)) :
        print("b has value ", end='')
        print(inParams.b, end='')
        print(" but expected to be ", end='')
        print("between ", end='')
        print(0.1, end='')
        print(" (d_min)", end='')
        print(" and ", end='')
        print(5.0, end='')
        print(" (d_max)", end='')
        print(".")
        raise Exception("InputError")
    if (not(4.5 <= inParams.w and inParams.w <= 910.0)) :
        print("w has value ", end='')
        print(inParams.w, end='')
        print(" but expected to be ", end='')
        print("between ", end='')
        print(4.5, end='')
        print(" (w_min)", end='')
        print(" and ", end='')
        print(910.0, end='')
        print(" (w_max)", end='')
        print(".")
        raise Exception("InputError")
    if (not(6.0 <= inParams.SD and inParams.SD <= 130.0)) :
        print("SD has value ", end='')
        print(inParams.SD, end='')
        print(" but expected to be ", end='')
        print("between ", end='')
        print(6.0, end='')
        print(" (SD_min)", end='')
        print(" and ", end='')
        print(130.0, end='')
        print(" (SD_max)", end='')
        print(".")
        raise Exception("InputError")
    if (not(inParams.AR <= 5.0)) :
        print("AR has value ", end='')
        print(inParams.AR, end='')
        print(" but expected to be ", end='')
        print("below ", end='')
        print(5.0, end='')
        print(" (AR_max)", end='')
        print(".")
        raise Exception("InputError")
    
    if (not(inParams.a > 0)) :
        print("a has value ", end='')
        print(inParams.a, end='')
        print(" but expected to be ", end='')
        print("above ", end='')
        print(0, end='')
        print(".")
        raise Exception("InputError")
    if (not(inParams.a >= inParams.b)) :
        print("a has value ", end='')
        print(inParams.a, end='')
        print(" but expected to be ", end='')
        print("above ", end='')
        print(inParams.b, end='')
        print(" (b)", end='')
        print(".")
        raise Exception("InputError")
    if (not(0 < inParams.b and inParams.b <= inParams.a)) :
        print("b has value ", end='')
        print(inParams.b, end='')
        print(" but expected to be ", end='')
        print("between ", end='')
        print(0, end='')
        print(" and ", end='')
        print(inParams.a, end='')
        print(" (a)", end='')
        print(".")
        raise Exception("InputError")
    if (not(inParams.w > 0)) :
        print("w has value ", end='')
        print(inParams.w, end='')
        print(" but expected to be ", end='')
        print("above ", end='')
        print(0, end='')
        print(".")
        raise Exception("InputError")
    if (not(0 <= inParams.P_btol and inParams.P_btol <= 1)) :
        print("P_btol has value ", end='')
        print(inParams.P_btol, end='')
        print(" but expected to be ", end='')
        print("between ", end='')
        print(0, end='')
        print(" and ", end='')
        print(1, end='')
        print(".")
        raise Exception("InputError")
    if (not(inParams.TNT > 0)) :
        print("TNT has value ", end='')
        print(inParams.TNT, end='')
        print(" but expected to be ", end='')
        print("above ", end='')
        print(0, end='')
        print(".")
        raise Exception("InputError")
    if (not(inParams.SD > 0)) :
        print("SD has value ", end='')
        print(inParams.SD, end='')
        print(" but expected to be ", end='')
        print("above ", end='')
        print(0, end='')
        print(".")
        raise Exception("InputError")
    if (not(inParams.AR >= 1)) :
        print("AR has value ", end='')
        print(inParams.AR, end='')
        print(" but expected to be ", end='')
        print("above ", end='')
        print(1, end='')
        print(".")
        raise Exception("InputError")


